src/NuGet.Core/NuGet.Commands/SourcesCommands/SourceRunners.cs

```diff
@@ -144,6 +156,24 @@ public static void Run(ListSourceArgs args, Func<ILogger> getLogger)
                                 source.Name,
                                 source.IsEnabled ? string.Format(CultureInfo.CurrentCulture, Strings.SourcesCommandEnabled) : string.Format(CultureInfo.CurrentCulture, Strings.SourcesCommandDisabled)));
                             getLogger().LogMinimal(string.Format("{0}{1}", sourcePadding, source.Source));
+
+                            if (source.IsHttp && !source.IsHttps)
+                            {
+                                if (httpPackageSources == null)
+                                {
+                                    httpPackageSources = new();
+                                }
+                                httpPackageSources.Add(source);
+                            }
+                        }
+
+                        if (httpPackageSources != null)
+                        {
+                            getLogger().LogWarning(
+                                string.Format(CultureInfo.CurrentCulture,
```

Comment:
I think there are 2 different scenarios possible here. A single HTTP source or multiple HTTP sources. The following implementation has different messages for each case. Can we reuse the same functionality here?

https://github.com/NuGet/NuGet.Client/blob/d970aa3fd6d607642ceb0905498ec0d75a03562c/src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackage/ListPackageCommandRunner.cs#L180-L198